roxctl/central/db/generate/generate_test.go

```diff
@@ -0,0 +1,406 @@
+		c := testCase
+		cfg = renderer.Config{}
+		s.Run(c.description, func() {
+			s.Require().NoError(os.RemoveAll(defaultCentralDBBundle), os.RemoveAll(testOutputDir))
+			rootCmd := s.createRootCommand(c)
+			cmd, output, errOut, err := executeCommand(rootCmd, c.args...)
+			s.Assert().NotNil(cmd)
+			s.Assert().Equal(c.cmdUse, cmd.Use)
+			if c.args[len(c.args)-1] == "-h" || c.checkUsage {
+				_, cmds, flags := s.parseUsage(output)
+				for _, contain := range c.containsCommands {
+					s.Assert().Contains(cmds, contain)
+				}
+				for _, flag := range c.containsFlags {
+					s.Assert().Contains(flags, flag)
+				}
+				if c.allCommands != nil {
+					for _, contain := range c.allCommands {
+						s.Assert().Contains(cmds, contain)
+					}
+					s.Assert().Len(cmds, len(c.allCommands))
+				}
+			}
+			if c.errContains != "" {
+				s.Assert().Error(err)
+				s.Assert().Contains(errOut, c.errContains)
+			} else {
+				s.Assert().Empty(errOut)
+				s.Assert().NoError(err)
+			}
+		})
+	}
+}
+
+func (s *centralDBGenerateCliTestSuite) createRootCommand(testCase testCaseType) *cobra.Command {
+	genCmd := Command(env.CLIEnvironment())
+	common.PatchPersistentPreRunHooks(genCmd)
+	if testCase.addError == downloadError {
+		genCmd.PersistentPreRunE = func(*cobra.Command, []string) error {
+			return errors.New("download error")
+		}
+	} else {
+		genCmd.PersistentPreRunE = func(*cobra.Command, []string) error {
+			cfg.SecretsByteMap = map[string][]byte{
+				"ca.pem":              []byte("ca.pem"),
+				"central-db-cert.pem": []byte("central-db-cert.pem"),
+				"central-db-key.pem":  []byte("central-db-key.pem"),
+				"central-db-password": []byte("password"),
+			}
+			return nil
+		}
+	}
+
+	s.setRunE(testCase, genCmd, "k8s", "pvc")
+	s.setRunE(testCase, genCmd, "k8s", "hostpath")
+	s.setRunE(testCase, genCmd, "k8s", "none")
+	s.setRunE(testCase, genCmd, "openshift", "pvc")
+	s.setRunE(testCase, genCmd, "openshift", "hostpath")
+	s.setRunE(testCase, genCmd, "openshift", "none")
+
+	return genCmd
+}
+
+func (s *centralDBGenerateCliTestSuite) setRunE(testCase testCaseType, root *cobra.Command, subCmds ...string) {
+	command := root
+	for _, subCmd := range subCmds {
+		command = s.lookUpCommand(command.Commands(), subCmd)
+	}
+	runE := command.RunE
+	command.RunE = func(cmd *cobra.Command, args []string) error {
+		// Run original command but tolerate the error.
+		_ = runE(cmd, args)
+		if testCase.addError == renderError {
+			return errors.New("render error")
+		}
+		s.verifyConfig(testCase, &cfg)
+		return nil
+	}
+}
+
+func (s *centralDBGenerateCliTestSuite) lookUpCommand(cmds []*cobra.Command, target string) *cobra.Command {
+	for _, cmd := range cmds {
+		if cmd.Use == target {
+			return cmd
+		}
+	}
+	s.Require().Fail("Cannot find command %s", target)
+	return nil
+}
+
+func (s *centralDBGenerateCliTestSuite) parseUsage(output string) (usage string, commands map[string][]string, flags map[string][]string) {
```

Comment:
Why do we need to parse usage? Why can't we just compare output with expected output? This will simplify the test and make it clear what if failing